### PR TITLE
fix: [M3-8590] - Restrict access to Database Create page for restricted users

### PR DESCRIPTION
**Pull request by zaenab-akamai**
## Description 📝
Discovered during release testing for v1.128.0:

A recent PR updated the UI for users who have restricted access (read only or no access) to Databases: https://github.com/linode/manager/pull/10794

This included disabling buttons and providing helper tooltip text to explain the reason why restricted users couldn't access these features.

The PR did not include the same treatment for the databases/create page, which a restricted user can still navigate to via the URL (https://cloud.linode.com/databases/create) or via the top menu's Create button. The restricted user is able to fill out the form and submit it, and only then do they receive an error.

Thank you @mjac0bs for your inputs! :)

## Changes  🔄
List any change relevant to the reviewer.
- Added an error message on the create database page for restricted users
- Disabled all controls on /databases/create
- Updated DatabaseCreateAccessControls to disable inputs through a prop

## Target release date 🗓️
NA

## Preview 📷

Screenshot 2024-10-23 at 4 29 20 PM

Screenshot 2024-10-23 at 4 29 30 PM

### Reproduction steps
1. Login as a user with full access and add databases. Grant read_only permissions to these new resources to the restricted user
2. Login as the restricted user and navigate to the databases page. The "Create Database Cluster" button should be disabled, but if you go to directly to https://cloud.linode.com/databases/create via the URL or via the top menu, notice that the form is accessible.

### Verification steps
Pull the PR, login as a restricted.
Navigate to /databases/create via the URL or via the top menu. The form should be disabled and an error notice should appear on top.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

---

### CODE REVIEW

**Review by dwiley-akamai — APPROVED** [review]
Access to Database Create page restricted appropriately for restricted users with no adverse impact on non-restricted users ✅ 

We'll want to get a [changeset added for this PR](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)

**Review by mjac0bs — APPROVED** [review]
Looks good - error message is visible for restricted users and form fields are disabled.

Other than Dajahi's comments, this should be good to go. Thanks @zaenab-akamai!

**Comment by zaenab-akamai** [comment]
Added the changeset. I missed this - thanks for pointing it out @dwiley-akamai !  
@mjac0bs I marked it as a fix since this was meant to fix a bug on a change released earlier. Let me know if you still prefer `Changed`. I'll update the PR title as well to keep things consistent

File: packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx [hunk 1]
```diff
@@ -570,6 +590,8 @@ const DatabaseCreate = () => {
             }}
             className={classes.selectPlanPanel}
             data-qa-select-plan
+            disabled={isRestricted}
+            disabledTabs={isRestricted ? ['shared', 'dedicated'] : []}
```
[the remark this answers is not in the document]
Comment (zaenab-akamai) [reply packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx]: Updated

**Comment by mjac0bs** [comment]
> @mjac0bs I marked it as a fix since this was meant to fix a bug on a change released earlier. Let me know if you still prefer `Changed`. I'll update the PR title as well to keep things consistent

@zaenab-akamai Good call on Fixed since there was a previous PR for this. That works for me. 👍🏼

**Pull request merged**